src/capi.rs

```diff
@@ -93,10 +93,23 @@ pub unsafe extern "C" fn microvmi_read_physical(
     physical_address: uint64_t,
     buffer: *mut uint8_t,
     size: size_t,
+    bytes_read: *mut uint64_t,
 ) -> bool {
     let driver = get_driver_mut_ptr(context);
+    let mut ignored_bytes_read: u64 = 0;
+    // if bytes_read param is NULL, just ignore the bytes_read in this local variable
+    let bytes_read_ptr = if bytes_read.is_null() {
```

Comment:
nit: This is a little awkward from a design perspective. Why not using a local var for the `read_physical` call and assigning the value to `bytes_read` afterwards (if it is not null)? This way you have one pointer var less and you don't need to conditionally assign pointers to other pointers.

Reply:
You're right, i refactored the code, it's much clearer now.